### ISSUE TITLE
Statusline counts for each split

### ISSUE THREAD

**Issue by nyrkovalex** [w0rp/ale#682]
Hi, is it possible to display individual error counts for each split without vim-airline?

Consider the following case:
* We have two vertical splits. 
* The one to the left has 3 erorrs and 0 warnings 
* The one to the right  has 0 errors and 2 warnings.

If we incorporate `ale#statusline#Count(bufnr(''))` output in the `statusline` we'll see same counts for each split's statusline. 

E.g. if we focus left split both statuslines will show `[3e 0w]` and if we focus the right one again both will show `[0e 2w]` or something like that depending on how we format the output.

_The question is whether it is possible to show counts corresponding to each split's buffer?_

The desired result regarding the case above is to always see `[3e 0w]` for the left buffer and `[0e 2w]` for the right.

Thank you.

**Comment by w0rp** [comment]
I recommend writing your own status function, as described here: https://github.com/w0rp/ale#faq-statusline

I tried to come up with some kind of format option which would let everyone do what they want, but in the end I decided it was so complicated that it was impossible, outside of basically writing a new DSL just for formatting status lines.

**Comment by nyrkovalex** [comment]
Thank you for the response.
That's what I'm trying to do. I just cannot figure out how to get a `buffnr` of an inactive (unfocused) buffer. So I decided to raise in issue hoping that someone from community might have already solved that :)
This particular case does not seem very googleable.

**Comment by w0rp** [comment]
Check out `:help getwininfo()`. You can list the buffers in a window. There's also `gettabinfo()`, which you can use to list the windows in a tab.

**Comment by nyrkovalex** [comment]
@w0rp Thank you, I'll look into it.

**Comment by nyrkovalex** [comment]
My initial intent was to have individual error and/or warning counts for each split with individual coloring: green for OK, red for errors, yellow for warnings.
I've figured out what I was doing wrong.

I used a `set statusline=%!StatusLine()` expression to create statusline and `%!`-thing eveluates in a context of _window_, so calling `ale#statusline#Count(bufnr(''))` inside my `StatusLine()` function always returned counts for an active buffer.
But `%!` result can incorporate `%{...}` calls which are evaluated in a context of a buffer being processed.
So the solution was to create separate functions for Ok, Error and Warning linter reports and call them via `%{...}`.

Something like
```
function! StatusLine() abort
  let l:lintermsg = '%#CheckedStatus#%{LinterOk()}' . '%#ErrorMsg#%{LinterErrors()}' . '%#WarningStatus#%{LinterWarnings()}%*'
  return '%f %m ' . l:lintermsg . '%= %{getcwd()} %#GitStatus#|%{fugitive#head(8)}|%* %l:%v'
endfunction
```

And it gives nice coloring while preserving individual counts for each buffer
![screenshot from 2017-06-23 21-30-22](https://user-images.githubusercontent.com/1652654/27495597-34367236-585b-11e7-855e-bf56c05ff0d2.png)
